qiskit/backends/basebackend.py

```diff
@@ -41,7 +41,7 @@ def __init__(self, configuration=None):
         Raises:
             FileNotFoundError if backend executable is not available.
         """
-        self._configuration = configuration or {}
+        self._configuration = configuration
```

Comment:
@diego-plan9 I think i did this because I *wanted* an error raised if there was no configuration backend or if the configuration backend did not conform to the schema (e.g. did not have a 'name') field. So i want to ensure the configuration is valid at instantiation. So I guess you mean we should remove the default of `None`?

Reply:
Yes - removing the default of `None` plus having a basic check in place (basically just make sure that `configuration['name']` can be retrieved) sounds good! The idea would be to make sure that all the methods of the base class do not raise an error if the instantiation was successful. 

Also, can you fix a small typo at  `__str__` (`_configuration`, with the underscore - or actually just use `self.name` directly)?